applications/debug/event_loop_blink_test/event_loop_blink_test.c

```diff
@@ -0,0 +1,165 @@
+#include <furi.h>
+#include <furi_hal_resources.h>
+
+#include <gui/gui.h>
+#include <gui/elements.h>
+#include <gui/view_port.h>
+
+#include <input/input.h>
+
+#define TAG "EventLoopBlinkTest"
+
+#define TIMER_COUNT (6U)
+
+typedef struct {
+    FuriEventLoop* event_loop;
+    FuriMessageQueue* input_queue;
+    FuriEventLoopTimer* timers[TIMER_COUNT];
+} EventLoopBlinkTestApp;
+
+static const GpioPin* blink_gpio_pins[] = {
+    &gpio_ext_pa7,
+    &gpio_ext_pa6,
+    &gpio_ext_pa4,
+    &gpio_ext_pb3,
+    &gpio_ext_pb2,
+    &gpio_ext_pc3,
+};
+
+static_assert(COUNT_OF(blink_gpio_pins) == TIMER_COUNT);
+
+static const uint32_t timer_intervals[] = {
+    320,
+    750,
+    1000,
+    500,
+    150,
+    210,
+};
+
+static_assert(COUNT_OF(timer_intervals) == TIMER_COUNT);
+
+static void blink_gpio_init(void) {
+    for(size_t i = 0; i < TIMER_COUNT; ++i) {
+        furi_hal_gpio_init_simple(blink_gpio_pins[i], GpioModeOutputPushPull);
+    }
+
+    furi_hal_gpio_init_simple(&gpio_ext_pc0, GpioModeOutputPushPull);
+}
+
+static void blink_gpio_deinit(void) {
+    for(size_t i = 0; i < TIMER_COUNT; ++i) {
+        furi_hal_gpio_init_simple(blink_gpio_pins[i], GpioModeAnalog);
+    }
+
+    furi_hal_gpio_init_simple(&gpio_ext_pc0, GpioModeAnalog);
+}
+
+static void view_port_draw_callback(Canvas* canvas, void* context) {
+    UNUSED(context);
+    canvas_clear(canvas);
+    elements_text_box(
+        canvas,
+        0,
+        0,
+        canvas_width(canvas),
+        canvas_height(canvas),
+        AlignCenter,
+        AlignCenter,
+        "\e#Event Loop Timers Test\e#\n"
+        "Press buttons\n"
+        "to enable or disable timers\n"
+        "\e#Exit\e# = long press \e#Back\e#",
+        false);
+}
+
+static void view_port_input_callback(InputEvent* input_event, void* context) {
+    EventLoopBlinkTestApp* app = context;
+    furi_message_queue_put(app->input_queue, input_event, 0);
+}
+
+static bool input_queue_callback(FuriMessageQueue* queue, void* context) {
+    EventLoopBlinkTestApp* app = context;
+
+    InputEvent event;
+    FuriStatus status = furi_message_queue_get(queue, &event, FuriWaitForever);
```

Comment:
As someone who is trying to study the usage of the event loop - why? This call is executed when there is something on the queue, so it should always immediately get some data.

Or is this to account for the cases of type "if this `queue_get` fails, then something is very wrong and you should crash instead of blocking endlessly"?

Reply:
we want to know if event loop was correctly designed, so we can fail early, FuriWaitForever will hide it from us

Reply:
> I think we should prevent people from using Forever if queue uses event loop.

True, but it is not trivial. I propose we do it in the next phase where we stabilise the primitive API.